### ISSUE TITLE
vmhgfs-fuse, umask paremater does not take effect

### ISSUE THREAD

**Issue by ls-zhu** [vmware/open-vm-tools#223]
Windows host shares a directory with Linux guest, then in Linux Guest, I run:

/usr/bin/vmhgfs-fuse -o subtype=vmhgfs-fuse -o uid=1000 -o umask=0600 /mnt/.

It expect that the shared folder under the mount point has 600 permissions. But we see:

localhost:/mnt # ll
total 0
d--xrwxrwx 1 lszhu root 0 Jan 16 14:23 test_dir
localhost:/mnt # cd test_dir/
localhost:/mnt/test_dir # ll
total 0
---xrwxrwx 1 lszhu root 0 Jan 16 14:23 1.txt

Thanks!

**Comment by lousybrit** [comment]
Looks like we used to mount options fmask and dmask  for the vmhgfs kernel client mount operation but these have not been migrated to or have their equivalent implemented in the FUSE client.

I will look into an fixing the missing mask, umask for the FUSE client.

Thanks for reporting.
Steve

**Comment by herb-kuta-colortokens** [comment]
> have their equivalent implementation in the FUSE client.

I see that even the latest released version (v3.2.1) of [fuse(8)](https://github.com/libfuse/libfuse/blob/fuse-3.2.1/doc/mount.fuse.8) doesn't mention dmask or fmask. Will it still be possible for vmhgfs-fuse to implement them?

**Comment by lousybrit** [comment]
I have this on my plate and some work is in progress, just on a temporary hold.
It is just behind some other work I need to complete before I can get back to it.

Steve

**Comment by feklee** [comment]
I don’t understand this issue. As far as I can tell `umask` works fine. (what doesn’t work is `fmask` and `dmask`, but that’s for another issue)

> Windows host shares a directory with Linux guest, then in Linux Guest, I run:
> 
> /usr/bin/vmhgfs-fuse -o subtype=vmhgfs-fuse -o uid=1000 -o umask=0600 /mnt/.
> 
> It expect that the shared folder under the mount point has 600 permissions.

No, you should expect files in the mount point to have permissions *except* those specified in the mask.

> But we see:
> 
> localhost:/mnt # ll
> total 0
> d--xrwxrwx 1 lszhu root 0 Jan 16 14:23 test_dir
> localhost:/mnt # cd test_dir/
> localhost:/mnt/test_dir # ll
> total 0
> ---xrwxrwx 1 lszhu root 0 Jan 16 14:23 1.txt

Looks correct to me. Files are lacking permissions 600. That’s as expected.

*Am I missing something?*
